main/zoe/api/Zoe-Intro.md

```diff
@@ -0,0 +1,109 @@
+# Introduction to Zoe
+
+
+**Note:** Before reading this, you should be familiar with ERTP basics. (**tyg todo**: Add link)
+
+
+This is an introduction to Zoe, Agoric's smart contract framework. The Zoe Guide (**tyg todo**: Add link) has more complete and detailed information, including concepts not covered here. In particular, here we simplify Zoe examples such that there are always two parties in a basic trading/swap contract. We do not discuss more complicated contracts with many participants, such as auctions, or simpler contracts, such as our AutomaticRefund one party demo contract.
+
+
+## What is Zoe?
+
+
+*Zoe* is itself a long lived and trusted contract that serves as a framework which runs other contracts.
+
+
+## How do Zoe contracts work?
+
+
+To start using a contract, you need to *install* it on Zoe. This is putting the contract’s code into Zoe. Multiple contracts can be installed on Zoe, and each contract can have multiple simultaneous active offers, each of which is handled by a separate contract instance. For clarity, in this document a *contract* is the installed code that defines how the contract works. A *contract instance* is an active execution of the contract code.
+
+
+Once installed on Zoe, a contract waits for an *offer* to be made. That starts a lifecycle for the offer and new contract instance of:
+1. Sending out *invites* to request counter-offers.
+2. Partie *redeeming* those invites and making a counter-offer.
+3. Determining when the offer *matches* a counter-offer.
+4. Either succeeding or failing to make the transaction. 
+
+
+An offer’s completion only affects that offer and the assets involved in it; essentially its contract instance. The contract it was made on keeps running, waiting for and processing new offers. 
+
+
+*Offer safety* is automatically enforced for parties in a contract; if an offer succeeds and completes, all parties get what they wanted. If it fails to complete successfully for any reason, parties keep what they offered.
```

Comment:
Better to say "parties are refunded what they offered". Saying "keep" implies that they never lost possession of it.

Reply:
Will rephrase, thanks.

Reply:
I think this will give the reader a wrong assumption that will be very hard to change later. An offer can be a perfectly valid offer, and the contract can still choose to give a refund for that offer, not what the user wanted. 

> if an offer succeeds and completes, all parties get what they wanted

This is generally not a true if/then. It really depends on the contract whether or not a user gets what they wanted or whether the user gets a refund for a valid offer. For an invalid offer (one that is not what the contract expects, but which was at least successfully escrowed with Zoe), the user will usually get a refund because the contract will complete the offer immediately.

> If it fails to complete successfully for any reason, parties keep what they offered.

There's a weird connection here between whether a single offer is completed by the contract with a refund allocation, and whether all parties get a refund. There's no such connection generally. If we want to talk about the cases in which any offer being rejected by a contract results in everyone getting a refund, we should use a specific contract example in this guide.